edx_proctoring/views.py

```diff
@@ -1104,3 +1107,50 @@ def post(self, request, user_id):  # pylint: disable=unused-argument
                         code = 500
                 seen.add(backend_name)
         return Response(data=results, status=code)
+
+
+class UserRetirement(AuthenticatedAPIView):
+    """
+    Retire user personally-identifiable information (PII) for a user
+    """
+    def _retire_exam_attempts_user_info(self, user_id):
+        """ Remove PII for exam attempts and exam history """
+        attempts = ProctoredExamStudentAttempt.objects.filter(user_id=user_id)
+        if attempts:
+            for attempt in attempts:
+                attempt.student_name = ''
+                attempt.last_poll_ipaddr = None
+                attempt.save()
+
+        attempts_history = ProctoredExamStudentAttemptHistory.objects.filter(user_id=user_id)
+        if attempts_history:
+            for attempt_history in attempts_history:
+                attempt_history.student_name = ''
+                attempt_history.last_poll_ipaddr = None
+                attempt_history.save()
+
+    def _retire_user_allowances(self, user_id):
+        """ Clear user allowance values """
+        allowances = ProctoredExamStudentAllowance.objects.filter(user=user_id)
+
+        if allowances:
```

Comment:
Nit: this check isn't necessary (ditto the one below) - every QuerySet is iterable, even if it's empty.